Golden InSAR test failed: https://github.com/ASFHyP3/hyp3-testing/actions/runs/19053621595/job/54419191963

prod jobs: https://hyp3-api.asf.alaska.edu/jobs?user_id=hyp3_testing&name=hyp3-testing-wPGHB6O

test jobs: https://hyp3-test-api.asf.alaska.edu/jobs?user_id=hyp3_testing&name=hyp3-testing-wPGHB6O

```
=========================== short test summary info ============================
FAILED tests/test_insar.py::test_golden_insar - ***.compare.ComparisonFailure: 7 differences found!!

--------------------------------------------------------------------------------
/tmp/pytest-of-runner/pytest-0/main/S1AA_20210513T162108_20210525T162109_VVP012_INT80_G_ueF_09C1/S1AA_20210513T162108_20210525T162109_VVP012_INT80_G_ueF_09C1_los_disp.tif
/tmp/pytest-of-runner/pytest-0/develop/S1AA_20210513T162108_20210525T162109_VVP012_INT80_G_ueF_6A64/S1AA_20210513T162108_20210525T162109_VVP012_INT80_G_ueF_6A64_los_disp.tif
--------------------------------------------------------------------------------
Values are different.

Two data are not similar with confidence level 98.0% (actual 45%)

--------------------------------------------------------------------------------
/tmp/pytest-of-runner/pytest-0/main/S1AA_20210513T162108_20210525T162109_VVP012_INT80_G_ueF_09C1/S1AA_20210513T162108_20210525T162109_VVP012_INT80_G_ueF_09C1_unw_phase.tif
/tmp/pytest-of-runner/pytest-0/develop/S1AA_20210513T162108_20210525T162109_VVP012_INT80_G_ueF_6A64/S1AA_20210513T162108_20210525T162109_VVP012_INT80_G_ueF_6A64_unw_phase.tif
--------------------------------------------------------------------------------
Values are different.

Two data are not similar with confidence level 98.0% (actual 45%)

--------------------------------------------------------------------------------
/tmp/pytest-of-runner/pytest-0/main/S1AA_20210513T162108_20210525T162109_VVP012_INT80_G_ueF_09C1/S1AA_20210513T162108_20210525T162109_VVP012_INT80_G_ueF_09C1_vert_disp.tif
/tmp/pytest-of-runner/pytest-0/develop/S1AA_20210513T162108_20210525T162109_VVP012_INT80_G_ueF_6A64/S1AA_20210513T162108_20210525T162109_VVP012_INT80_G_ueF_6A64_vert_disp.tif
--------------------------------------------------------------------------------
Values are different.

Two data are not similar with confidence level 98.0% (actual 40%)

--------------------------------------------------------------------------------
/tmp/pytest-of-runner/pytest-0/main/S1AA_20200110T172104_20200203T172103_VVP024_INT80_G_weF_A6EE/S1AA_20200110T172104_20200203T172103_VVP024_INT80_G_weF_A6EE_unw_phase.tif
/tmp/pytest-of-runner/pytest-0/develop/S1AA_20200110T172104_20200203T172103_VVP024_INT80_G_weF_13AF/S1AA_20200110T172104_20200203T172103_VVP024_INT80_G_weF_13AF_unw_phase.tif
--------------------------------------------------------------------------------
Values are different.

Two data are not similar with confidence level 98.0% (actual 0%)

--------------------------------------------------------------------------------
/tmp/pytest-of-runner/pytest-0/main/S1AA_20200110T172104_20200203T172103_VVP024_INT80_G_weF_34CD/S1AA_20200110T172104_20200203T172103_VVP024_INT80_G_weF_34CD_los_disp.tif
/tmp/pytest-of-runner/pytest-0/develop/S1AA_20200110T172104_20200203T172103_VVP024_INT80_G_weF_C746/S1AA_20200110T172104_20200203T172103_VVP024_INT80_G_weF_C746_los_disp.tif
--------------------------------------------------------------------------------
Values are different.

Two data are not similar with confidence level 98.0% (actual 0%)

--------------------------------------------------------------------------------
/tmp/pytest-of-runner/pytest-0/main/S1AA_20200110T172104_20200203T172103_VVP024_INT80_G_weF_34CD/S1AA_20200110T172104_20200203T172103_VVP024_INT80_G_weF_34CD_unw_phase.tif
/tmp/pytest-of-runner/pytest-0/develop/S1AA_20200110T172104_20200203T172103_VVP024_INT80_G_weF_C746/S1AA_20200110T172104_20200203T172103_VVP024_INT80_G_weF_C746_unw_phase.tif
--------------------------------------------------------------------------------
Values are different.

Two data are not similar with confidence level 98.0% (actual 0%)

--------------------------------------------------------------------------------
/tmp/pytest-of-runner/pytest-0/main/S1AA_20200110T172104_20200203T172103_VVP024_INT80_G_weF_34CD/S1AA_20200110T172104_20200203T172103_VVP024_INT80_G_weF_34CD_vert_disp.tif
/tmp/pytest-of-runner/pytest-0/develop/S1AA_20200110T172104_20200203T172103_VVP024_INT80_G_weF_C746/S1AA_20200110T172104_20200203T172103_VVP024_INT80_G_weF_C746_vert_disp.tif
--------------------------------------------------------------------------------
Values are different.

Two data are not similar with confidence level 98.0% (actual 0%)
=================== 1 failed, 4 passed in 4839.40s (1:20:39) ===================
```

I downloaded the last pair of products, `34CD` and `C746`, and opened all the tifs in qgis. It looks like for the failing image pairs, there's a small patch of pixels that gets lighter or darker (depending on which pair of images are being compared) between test and prod. Otherwise the images look quite similar to me. I'm not sure if this difference is worth caring about, or if it's worth looking at the other failing job pairs.

--- a/.github/workflows/test-and-build.yml
+++ b/.github/workflows/test-and-build.yml
@@ -76,14 +76,14 @@ jobs:
             org.opencontainers.image.revision=${{ github.sha }}
 
       - name: Add test tag
-        if: github.ref == 'refs/heads/develop'
+        if: ${{ github.event_name != 'pull_request' && contains(inputs.version_tag, '.dev') }}
         uses: akhilerm/tag-push-action@v2.2.0
         with:
           src: ${{ env.HYP3_REGISTRY }}/${{ github.event.repository.name }}:${{ env.SDIST_VERSION }}
           dst: ${{ env.HYP3_REGISTRY }}/${{ github.event.repository.name }}:test
 
       - name: Add latest tag
-        if: github.ref == 'refs/heads/main'
+        if: ${{ github.event_name != 'pull_request' && ! contains(inputs.version_tag, '.dev') }}
         uses: akhilerm/tag-push-action@v2.2.0
         with:
           src: ${{ env.HYP3_REGISTRY }}/${{ github.event.repository.name }}:${{ env.SDIST_VERSION }}

--- a/.github/workflows/test-and-build.yml
+++ b/.github/workflows/test-and-build.yml
@@ -1,9 +1,5 @@
 name: Test and build
 
-concurrency:
-  group: ${{ inputs.version_tag }}
-  cancel-in-progress: true
-
 on:
   push:
     branches:
@@ -37,6 +33,9 @@ jobs:
   dockerize:
     needs: call-version-info-workflow
     runs-on: ubuntu-latest
+    concurrency:
+      group: ${{ needs.call-version-info-workflow.outputs.version_tag }}
+      cancel-in-progress: true
     env:
       SDIST_VERSION: ${{ needs.call-version-info-workflow.outputs.version_tag }}
     steps:

--- a/.github/workflows/test-and-build.yml
+++ b/.github/workflows/test-and-build.yml
@@ -1,5 +1,9 @@
 name: Test and build
 
+concurrency:
+  group: ${{ inputs.version_tag }}
+  cancel-in-progress: true
+
 on:
   push:
     branches:

